src/command_modules/azure-cli-eventgrid/azure/cli/command_modules/eventgrid/custom.py

```diff
@@ -284,7 +392,160 @@ def update_event_subscription(
     params = EventSubscriptionUpdateParameters(
         destination=event_subscription_destination,
         filter=event_subscription_filter,
-        labels=event_subscription_labels
+        labels=event_subscription_labels,
+        retry_policy=retry_policy,
+        dead_letter_destination=deadletter_destination
     )
 
     return params
+
+
+def _get_endpoint_destination(endpoint_type, endpoint):
+    if endpoint_type.lower() == WEBHOOK_DESTINATION.lower():
+        destination = WebHookEventSubscriptionDestination(endpoint_url=endpoint)
+    elif endpoint_type.lower() == EVENTHUB_DESTINATION.lower():
+        destination = EventHubEventSubscriptionDestination(resource_id=endpoint)
+    elif endpoint_type.lower() == HYBRIDCONNECTION_DESTINATION.lower():
+        destination = HybridConnectionEventSubscriptionDestination(resource_id=endpoint)
+    elif endpoint_type.lower() == STORAGEQUEUE_DESTINATION.lower():
+        destination = _get_storage_queue_destination(endpoint)
+
+    return destination
+
+
+def _get_storage_queue_destination(endpoint):
+    # Supplied endpoint would be in the following format:
+    # /subscriptions/.../storageAccounts/sa1/queueServices/default/queues/{queueName}))
+    # and we need to break it up into:
+    # /subscriptions/.../storageAccounts/sa1 and queueName
+    queue_items = re.split(
+        "/queueServices/default/queues/", endpoint, flags=re.IGNORECASE)
+
+    if len(queue_items) != 2 or queue_items[0] is None or queue_items[1] is None:
+        raise CLIError('Argument Error: Expected format of --endpoint for storage queue is:' +
+                       '/subscriptions/id/resourceGroups/rg/providers/Microsoft.Storage/' +
+                       'storageAccounts/sa1/queueServices/default/queues/queueName')
+
+    destination = StorageQueueEventSubscriptionDestination(
+        resource_id=queue_items[0], queue_name=queue_items[1])
+
+    return destination
+
+
+def _get_deadletter_destination(deadletter_endpoint):
+    blob_items = re.split(
+        "/blobServices/default/containers/", deadletter_endpoint, flags=re.IGNORECASE)
+
+    if len(blob_items) != 2 or blob_items[0] is None or blob_items[1] is None:
+        raise CLIError('Argument Error: Expected format of --deadletter-endpoint is:' +
+                       '/subscriptions/id/resourceGroups/rg/providers/Microsoft.Storage/' +
+                       'storageAccounts/sa1/blobServices/default/containers/containerName')
+
+    deadletter_destination = StorageBlobDeadLetterDestination(
+        resource_id=blob_items[0], blob_container_name=blob_items[1])
+
+    return deadletter_destination
+
+
+def _validate_retry_policy(max_delivery_attempts, event_ttl):
+    if max_delivery_attempts < 1 or max_delivery_attempts > 30:
+        raise CLIError('--max-delivery-attempts should be a number between 1 and 30.')
+
+    if event_ttl < 1 or event_ttl > 1440:
+        raise CLIError('--event-ttl should be a number between 1 and 1440.')
```

Comment:
recommend using a validator for this so your parameter validation is seperate from your custom command logic

Reply:
This was actually a validator. Anyhow, we can split it to more specific ones.